app/src/main/java/net/squanchy/settings/view/SettingsHeaderLayout.kt

```diff
@@ -38,18 +38,18 @@ class SettingsHeaderLayout(context: Context, attrs: AttributeSet?) : AppBarLayou
     private fun updateWithAuthenticatedUser(user: User) {
         val googleUserInfo = user.googleData
         if (googleUserInfo != null) {
-            updateUserPhotoFrom(googleUserInfo)
+            updateUserPhotoFrom(googleUserInfo, imageLoader)
             usernameTextView.text = googleUserInfo.displayName
         }
     }
 
-    private fun updateUserPhotoFrom(userInfo: GoogleData) {
+    private fun updateUserPhotoFrom(userInfo: GoogleData, imageLoader: ImageLoader?) {
         if (imageLoader == null) {
-            return
+            throw IllegalStateException("Unable to access the ImageLoader, it hasn't been initialized yet")
```

Comment:
`error`

Reply:
This is the same code we're using in the other views. ISE looks correct to me, you're acting on something that is not there in the current state

Reply:
Detekt _really_ hates when you throw a raw Error. Open an issue to track this, we can improve it later on.